test/connectors/UpscanInitiateConnectorSpec.scala

```diff
@@ -32,45 +32,78 @@ import utils.WireMockHelper
 
+  lazy val connector: UpscanInitiateConnector = application.injector.instanceOf[UpscanInitiateConnector]
+
+  override def beforeEach(): Unit = server.resetAll()
+
+  val url = "/upscan/v2/initiate"
+
   "initiate" must {
 
-    "must return a result when the server responds with OK" in {
-      val app = application
-      running(app) {
-
-        val url = "/upscan/v2/initiate"
-        val responseBody =
-          s"""{
-             |    "reference": "11370e18-6e24-453e-b45a-76d3e32ea33d",
-             |    "uploadRequest": {
-             |        "href": "https://xxxx/upscan-upload-proxy/bucketName",
-             |        "fields": {
-             |            "Content-Type": "application/xml",
-             |            "success_action_redirect": "https://myservice.com/nextPage",
-             |            "error_action_redirect": "https://myservice.com/errorPage"
-             |        }
-             |    }
-             |}""".stripMargin
-        val connector = app.injector.instanceOf[UpscanInitiateConnector]
-        server.stubFor(
-          post(urlEqualTo(url))
-            .willReturn(ok(responseBody))
-        )
-
-        val result = connector.initiate(UpscanInitiateRequest("/some-callback-url")).futureValue
-
-        result.reference mustBe "11370e18-6e24-453e-b45a-76d3e32ea33d"
+    "return a successful result when the server responds with OK" in {
+
+      val responseBody =
+        s"""{
+           |    "reference": "11370e18-6e24-453e-b45a-76d3e32ea33d",
+           |    "uploadRequest": {
+           |        "href": "https://xxxx/upscan-upload-proxy/bucketName",
+           |        "fields": {
+           |            "Content-Type": "application/xml",
+           |            "success_action_redirect": "https://myservice.com/nextPage",
+           |            "error_action_redirect": "https://myservice.com/errorPage"
+           |        }
+           |    }
+           |}""".stripMargin
+
+      server.stubFor(
+        post(urlEqualTo(url))
+          .willReturn(ok(responseBody))
+      )
+
+      val result = connector.initiate(UpscanInitiateRequest("/some-callback-url")).futureValue
+
+      result.reference mustBe "11370e18-6e24-453e-b45a-76d3e32ea33d"
+    }
+
+    "return a failure result when the server responds with 400" in {
+
+      val exMessage = "Bad request, try again"
+
+      server.stubFor(
+        post(urlEqualTo(url))
+          .willReturn(badRequest().withBody(exMessage))
+      )
+
+      lazy val thrown = intercept[UpscanInitiateError] {
+        await(connector.initiate(UpscanInitiateRequest("/some-callback-url")))
       }
+
+      thrown.getMessage must include(exMessage)
     }
-  }
 
+    "return a failure result when the server responds with 500" in {
+
+      val exMessage = "Bad data"
+
+      server.stubFor(
+        post(urlEqualTo(url))
+          .willReturn(serverError().withBody(exMessage))
+      )
+
+      lazy val thrown = intercept[UpscanInitiateError] {
```

Comment:
Lazy val in this place is not really necessary, because you're using those values just below.